components/server/src/ome/security/basic/BasicACLVoter.java

```diff
@@ -84,12 +89,27 @@
 
     protected final Roles roles;
 
+    /* ignored if empty set */
+    private Set<Class<? extends IObject>> chgrpPermittedClasses = Collections.emptySet();
+
+    /* ignored if empty set */
+    private Set<Class<? extends IObject>> chownPermittedClasses = Collections.emptySet();
+
     private final LightAdminPrivileges adminPrivileges;
 
+    /* preserve permissions past context invalidation through to post-processing */
+    private final ThreadLocal<Set<AdminPrivilege>> adminPrivilegesCache = new ThreadLocal<>();
```

Comment:
Thank you, noted on https://trello.com/c/HRJyHGHh/18-review-implementation-approach as something to at least come back to. Possibly the privileges should be part of the event context in the first place!